frontend/src/components/App.tsx

```diff
@@ -0,0 +1,116 @@
+import * as React from 'react';
+import { useContext, useEffect, useState } from 'react';
+import axios, { AxiosResponse } from 'axios';
+import { HapiPractitionersResponse } from '../types/Hapi.types';
+import PractitionerFeed from './PractitionerFeed';
+import { HapiPractitionersFetchRequestUrl } from '../constants';
+import { AppContext } from '../AppContext';
+import { NormalizedPractitioner } from '../utils/helpers';
+import { Input, Button, CircularProgress } from '@mui/material';
+
+function App() {
+  const appContext = useContext(AppContext);
+  const [practitioners, setPractitioners] = useState<NormalizedPractitioner[]>([]);
+  const [searchTerm, setSearchTerm] = useState('');
+
+  useEffect( () => {
+    let searchUrl = HapiPractitionersFetchRequestUrl;
```

Comment:
Having practitioner-related logic live in the main App.tsx feels a little wrong to me. Not a big deal, but I'd move much of this to a parent component. Could be a premature optimization, as we don't know that the app will grow in scope, but I like to keep my main App file clean.

Reply:
that's a good point! on previous projects i like to use an api wrapper / guard and do api calls and state management updates from there. but like you said i didn't sense it was an optimization i needed to do yet.